### DISCUSSION TITLE
Trying to obtain token

### DISCUSSION THREAD

**Discussion by J-shw** [ianjwhite99/connected-car-node-sdk#44] [Q&A]
I am trying to obtain a token to access the API to control vehicles. I have been trying to make my own auth flow but that was getting messy and I am very new to using oAuth.. 

I found this repo and have been messing around trying to get it to work. I know alot has changed with Fords API over the last few months, so I am not sure if this even works anymore.

I am trying to use this package with Electron, to build a simple dekstop app. This is the code I am using:
`const client = connectedcar.AuthClient('9fb503e0-715b-47e8-adfd-ad4b7770f73b', {region: 'EU'});
    const token = await client.getAccessTokenFromCredentials({
        username: credentials.usr,
        password: credentials.psd,
    });`

The console outputs:
`(node:24798) UnhandledPromiseRejectionWarning: Error: certificate has expired
   at TLSSocket.onConnectSecure (node:_tls_wrap:1659:34)
   at TLSSocket.emit (node:events:514:28)
   at TLSSocket._finishInit (node:_tls_wrap:1070:8)
   at ssl.onhandshakedone (node:_tls_wrap:856:12)
(node:24798) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch().`

Any help would be appreciated, thanks!

**Comment by ianjwhite99** [comment]
Hi,

Ford has changed a lot over the past few months. I've been fairly busy and haven't been able to make updates to this or the python SDK as I haven't had time to go piece by piece through Fords new Auth flow. So at the moment this library is not able to get the auth credentials to make requests to Fords APIs.

**Comment by J-shw** [comment]
Ah ok no worries.

I have been playing with node.js and trying to learn how OAuth works. By translating some python code I found from Home assistants fordpass integration ([https://github.com/itchannel/fordpass-ha](https://github.com/itchannel/fordpass-ha)) into java script 

But I am very new to OAuth so haven't really got much of a chance of getting it working.

Thanks anyway!
